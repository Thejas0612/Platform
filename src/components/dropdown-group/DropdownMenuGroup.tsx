import React from 'react';
import DropdownMenu from './DropdownMenu';
import { Option } from './DropdownMenu';



export interface DropdownData {
  id: string;
  label: string;
  options: Option[];
}

interface DropdownMenuGroupProps {
  dropdownsData: DropdownData[];
  onChange: (id: string, value: string) => void;
  
}

const DropdownMenuGroup: React.FC<DropdownMenuGroupProps> = ({ dropdownsData, onChange }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'row', gap: '10px', flexWrap: 'wrap', justifyContent: 'center' }}>
      {dropdownsData.map((dropdown) => (
        <div key={dropdown.id}  style={{
            flex: '1 1 200px', 
            minWidth: '100px', 
            maxWidth: '350px', 
            margin: '15px', 
          }}>
          <DropdownMenu
            id={dropdown.id}
            label={dropdown.label}
            onChange={onChange}
            options={dropdown.options}
          />
        </div>
      ))}
    </div>
  );
};

export default DropdownMenuGroup;


