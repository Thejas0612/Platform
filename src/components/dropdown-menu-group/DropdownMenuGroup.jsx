import React from "react";
import PropTypes from "prop-types";
import { Grid, Typography } from "@mui/material";
import { MultipleSelectInput } from "./MultipleSelectInput";
import { MsolComponentHighlighter } from "../msol-component-highlighter/MsolComponentHighlighter";

export const DropdownMenuGroup = ({ data, value, onChange }) => {
  function handleChange(e, dropId, dropVal){
    const newValue = {
      ...value,
      [dropId]: dropVal
    }
    onChange(e, undefined, "", newValue);
  }
    return (
        <MsolComponentHighlighter>
            <Grid container spacing={2}>
                <Grid item xs={12}>
                    <Typography variant="subtitle2">Filter By:</Typography>
                </Grid>
                {data.map((dropdown) => (
                    <Grid key={dropdown.id} item xs={6} md={4} xl={3}>
                        <MultipleSelectInput
                            {...dropdown}
                            value={value[dropdown.id]}
                            onChange={(event, value) => {
                                handleChange(event, dropdown.id, value);
                            }}
                        />
                    </Grid>
                ))}
            </Grid>
        </MsolComponentHighlighter>
    );
};

DropdownMenuGroup.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string.isRequired,
      placeholder: PropTypes.string.isRequired,
      options: PropTypes.arrayOf(PropTypes.shape({
        label: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired,
      })).isRequired,
    })).isRequired,
    value: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
    onChange: PropTypes.func,
};
