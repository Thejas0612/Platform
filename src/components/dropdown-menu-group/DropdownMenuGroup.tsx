import { FC, memo } from "react";
import { Grid, Typography } from "@mui/material";
import { MultipleSelectInput, MultipleSelectInputProps } from "./MultipleSelectInput";
import { MsolComponentHighlighter } from "../msol-component-highlighter/MsolComponentHighlighter";

export interface Dropdown extends Omit<MultipleSelectInputProps, "onChange"|"value"> {
  id: string;
}

export interface DropdownMenuGroupProps {
  dropdowns: Dropdown [];
  selectedOptions:{ [key: string]: string[] };
  onChange?: (dropdownId: string, value: string[]) => void;
}

export const DropdownMenuGroup: FC<DropdownMenuGroupProps> = memo(({
                                                                      dropdowns,
                                                                      selectedOptions,
                                                                      onChange = () => {
                                                                      }
                                                                    }) => {
  return <MsolComponentHighlighter>
    <Grid container spacing={2}>
      <Grid item xs={12}>
        <Typography variant="subtitle2">Filter By:</Typography>
      </Grid>

      {dropdowns.map((dropdown) => (
        <Grid key={dropdown.id} item xs={6} md={4} xl={3}>
          <MultipleSelectInput
            {...dropdown}
            value={selectedOptions[dropdown.id]}
            onChange={(value) => {
              onChange(dropdown.id, value);
            }} />
        </Grid>
      ))}
    </Grid>
  </MsolComponentHighlighter>;
});
