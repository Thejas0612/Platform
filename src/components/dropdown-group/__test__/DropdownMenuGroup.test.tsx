import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import DropdownMenuGroup, { DropdownData } from '../DropdownMenuGroup';


describe('<DropdownMenuGroup />', () => {
  const options: DropdownData[] = [
    {
      id: '"high"',
      label: 'HIGH ACCURACY',
      options: [
        { value: 'yes', label: 'Yes' },
        { value: 'no', label: 'No' }
      ]
    },
    {
      id: 'hygienic',
      label: 'HYGIENIC / SANITARY',
      options: [
        { value: 'option1', label: 'Option 1' },
        { value: 'option2', label: 'Option 2' }
      ]
    }
  ];

  

  it('When default state render 2 dropdowns ', ()=>{
    const onChangeMock = jest.fn();
    const {container} = render(
        <DropdownMenuGroup dropdownsData={options} onChange={onChangeMock} />
      );
      expect(container).toMatchSnapshot()
  })


  it('when selecting a high accuracy option, then check the high accuracy option', async() => {
    const onChangeMock = jest.fn();
    const {container} = render(
      <DropdownMenuGroup dropdownsData={options} onChange={onChangeMock} />
    );

    const dropdown = await screen.findAllByRole('combobox')
    const highDropdown = await dropdown[0]
    const hygienicDropdown = await dropdown[1]

    expect(highDropdown).toBeInTheDocument()
    expect(hygienicDropdown).toBeInTheDocument()

    fireEvent.mouseDown(highDropdown);
    const optionElement = await screen.findByText(options[0].options[0].label) 
    fireEvent.mouseDown(optionElement)
  
    expect(optionElement).toBeInTheDocument();
    expect(container).toMatchSnapshot()
    

  });
})