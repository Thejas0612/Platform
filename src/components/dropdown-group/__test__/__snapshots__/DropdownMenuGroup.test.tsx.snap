// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownMenuGroup /> When default state render 2 dropdowns  1`] = `
<div>
  <div
    style="display: flex; flex-direction: row; gap: 10px; flex-wrap: wrap; justify-content: center;"
  >
    <div
      style="flex: 1 1 200px; min-width: 100px; max-width: 350px; margin: 15px;"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          HIGH ACCURACY
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1kd47wj-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls="mui-1"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                HIGH ACCURACY
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      style="flex: 1 1 200px; min-width: 100px; max-width: 350px; margin: 15px;"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          HYGIENIC / SANITARY
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1kd47wj-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls="mui-2"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                HYGIENIC / SANITARY
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DropdownMenuGroup /> when selecting a high accuracy option, then check the high accuracy option 1`] = `
<div
  aria-hidden="true"
>
  <div
    style="display: flex; flex-direction: row; gap: 10px; flex-wrap: wrap; justify-content: center;"
  >
    <div
      style="flex: 1 1 200px; min-width: 100px; max-width: 350px; margin: 15px;"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          HIGH ACCURACY
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl  css-1kd47wj-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls="mui-3"
            aria-expanded="true"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined MuiSelect-iconOpen css-bpeome-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                HIGH ACCURACY
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      style="flex: 1 1 200px; min-width: 100px; max-width: 350px; margin: 15px;"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          HYGIENIC / SANITARY
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1kd47wj-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-controls="mui-4"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                HYGIENIC / SANITARY
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
